src/Development/IDE/Spans/Documentation.hs

```diff
@@ -172,7 +172,10 @@ lookupHtmlForModule :: (FilePath -> FilePath -> FilePath) -> DynFlags -> Module
 lookupHtmlForModule mkDocPath df m = do
   let mfs = go <$> (listToMaybe =<< lookupHtmls df ui)
   htmls <- filterM doesFileExist (concat . maybeToList $ mfs)
-  return $ listToMaybe htmls
+  -- canonicalize located html to remove /../ indirection which can break some clients
+  -- (vscode on Windows at least)
+  htmls' <- traverse canonicalizePath htmls
+  return $ listToMaybe htmls'
```

Comment:
This code applies doesFileExist/canonicalizePath to a sequence of items, then only takes the first. Yuk. That means we are paying for more IO operations that we need. Can we change the `filterM` to be a `findM`, then we have a `Maybe FIlePath`, you can still traverse it to canonicalise, but now we only canonicalize one element, and can skip the `listToMaybe`.

(I appreciate this wasn't your code to begin with, but a good thing to fix)

Reply:
Did only the easy change so far, will look into testability a bit later
Thanks for the hint re findM, didn't know it (will take it as reminder to use Hoogle more often). I also had a thought about potentially wasteful IO on file exists/canonicalize , my naive intuition was that laziness will take care of it (which is not the case as it turned out)